pkg/rds.go

```diff
@@ -586,22 +588,32 @@ func (e *RDSExporter) addAllInstanceMetrics(sessionIndex int, instances []*rds.D
 }
 
 // Determines status from the number of days until EOL
-func getEOLStatus(eol string) (string, error) {
+func (e *RDSExporter) getEOLStatus(eol string, thresholds []Thresholds) (string, error) {
 	eolDate, err := time.Parse("2006-01-02", eol)
 	if err != nil {
 		return "", err
 	}
+	highestThreshold := 0
+	highestThresholdName := ""
 	currentDate := time.Now()
 	daysToEOL := int(eolDate.Sub(currentDate).Hours() / 24)
-	eolStatus := ""
-	if daysToEOL < 90 {
-		eolStatus = "red"
-	} else if daysToEOL < 180 {
-		eolStatus = "yellow"
-	} else {
-		eolStatus = "green"
+
+	for _, threshold := range thresholds {
+		if daysToEOL < threshold.Threshold {
+			return threshold.Name, nil
+		}
+		if threshold.Threshold > highestThreshold {
+			highestThreshold = threshold.Threshold
+			highestThresholdName = threshold.Name
+		}
```

Comment:
`thresholds` may not order by `Threshold`, to follow the existing logic,
need to first sort by `Threshold`, then return the first name when daysToEOL less than threshold, if not found, return the last name.

Need to add test cases for this.

Reply:
some edge cases:

* `daysToEOL` is exactly `thresholds[len(thresholds)-1].Days`
* can `thresholds` be empty?

The expected return value can't reach `"", nil`, and the log is never executed, right?

Reply:
Good catch, thresholds should not ever be empty but I added error handling just in case